x/stakeibc/keeper/msg_server_create_trade_route.go

```diff
@@ -107,20 +109,36 @@ func (ms msgServer) CreateTradeRoute(goCtx context.Context, msg *types.MsgCreate
 		ToAccount:         hostICA,
 	}
 
+	// If a max allowed swap loss is not provided, use the default
+	maxAllowedSwapLossRate := msg.MaxAllowedSwapLossRate
+	if maxAllowedSwapLossRate == "" {
```

Comment:
This is nice, I think we should do the same thing for min_swap_amount and max_swap_amount defaulting to 0 and maxint64 if they come in as empty strings

Reply:
they're actually int types so they default to 0 if not provided!

Reply:
ok nice, but that isn't good for the max swap size, it would cause all the swaps to exit before sending ICAs right now

Reply:
we have a validate basic check that confirms max > min, so I don't think a max of 0 is valid

Reply:
So then if they do not include any input value for min and max it will actually not be legal whereas the reasonable defaults feel like no min no max and no max would be implemented as a default value of maxint64 right?  I think we need a similar if statement to set this as the max swap default

Reply:
tis a good point

Reply:
should we do max int or just leave it as 0 and only do the check if if it's != 0?

Reply:
I would do max int so that it exists and the normal checks just work out of the box

Reply:
I'm realizing it might be an unintuitive UX to query the trade route and see 18446744073709551615

Reply:
whoops, meant to send that before your last message. Does that change your stance at all?

Reply:
also, I think if we do math.MaxUint, that'll still be a tiny number of tokens for host zones like injective/evmos with a more granular micro denom

Reply:
This also gives a way via update tx that humans can pause the transfers and the swapping.

If they set the min_swap to be a very high number, new funds would stop being transferred over to the trade route and if they set the max_swap to 0 they have effectively paused swaps from happening

I know there are checks that min_swap < max_swap but maybe we don't actually want or need those because setting it such that min_swap > max_swap is actually useful to pause activity in this way

Reply:
You could make it the more readable 10e19 for humans instead of the absolute max

Reply:
is there any reason we'd do that to pause instead of just removing the trade route?

Reply:
I understand your 0 explicit check solution and I think that would be fine if we don't think that the max actually being zero could be a useful case

Reply:
You are right we would probably just remove the trade route.  It could be something very niche, like update the min swap to stop new tranfers, wait 2-3 stride epochs for the current funds in flight in the system to be traded and returned to host, then remove the route.  It allows you to stop the transfers and swaps independently but I can't imagine a real usecase where you would want transfers into the trade ICA building up without swapping happening can you?

Reply:
i can't either, but I'm onboard with just doing a human readable number like 10e24

Reply:
I think we also should change the type to string cause max's for host zone's with 18 decimal tokens will exceed the uint limit

Reply:
changed the type to sdk.Int in latest commit